graph-builder/src/registry.rs

```diff
@@ -262,6 +274,34 @@ pub fn fetch_releases<S: ::std::hash::BuildHasher>(
             .metadata
             .insert(manifestref_key.to_owned(), manifestref);
 
+        match manifest.architectures() {
+            Ok(archs) => {
+                let arch = {
+                    ensure!(
+                        archs.len() == 1,
+                        "[{}] broke assumption of exactly one architecture per tag: {:?}",
+                        tag,
+                        archs
+                    );
+                    archs.first().unwrap().to_string()
+                };
+
+                // Encode the architecture as SemVer information
+                release.metadata.version.build =
+                    vec![semver::Identifier::AlphaNumeric(arch.clone())];
+
+                // Attach the architecture for later processing
+                release
+                    .metadata
+                    .metadata
+                    .insert("io.openshift.upgrades.graph.release.arch".to_owned(), arch);
```

Comment:
Should we be using the constants from the arch filter's `key_prefix` and `key_suffix` (both of which are `pub`, right?) instead of hard-coding `io.openshift.upgrades.graph.release.arch`?